Streetcode/Streetcode.XUnitTest/MediatRTests/Teams/GetAllTeamsTest.cs

```diff
@@ -0,0 +1,137 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.Team;
+using Streetcode.BLL.MediatR.Team.GetAll;
+using Streetcode.DAL.Entities.Team;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Xunit;
+
+namespace Streetcode.XUnitTest.MediatRTests.Teams;
+
+public class GetAllTeamTest
+{
+    private Mock<IRepositoryWrapper> _mockRepository;
+    private Mock<IMapper> _mockMapper;
+
+    public GetAllTeamTest()
+    {
+        _mockRepository = new Mock<IRepositoryWrapper>();
+        _mockMapper = new Mock<IMapper>();
+    }
+
+    [Fact]
+    public async Task ShouldReturnSuccessfully_CorrectType()
+    {
+        //Arrange
+        _mockRepository.Setup(x => x.TeamRepository.GetAllAsync(
+            null,
+            It.IsAny<Func<IQueryable<TeamMember>, IIncludableQueryable<TeamMember, object>>>()))
+            .ReturnsAsync(GetTeamList());
+
+        _mockMapper
+            .Setup(x => x.Map<IEnumerable<TeamMemberDTO>>(It.IsAny<IEnumerable<TeamMember>>()))
+            .Returns(GetListTeamDTO());
+
+        var handler = new GetAllTeamHandler(_mockRepository.Object, _mockMapper.Object);
+
+        //Act
+        var result = await handler.Handle(new GetAllTeamQuery(), CancellationToken.None);
+
+        //Assert
+        Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.IsType<List<TeamMemberDTO>>(result.ValueOrDefault)
+        );
+    }
+
+    [Fact]
+    public async Task ShouldReturnSuccessfully_CountMatch()
+    {
+        //Arrange
+        _mockRepository.Setup(x => x.TeamRepository.GetAllAsync(
+            null,
+            It.IsAny<Func<IQueryable<TeamMember>, IIncludableQueryable<TeamMember, object>>>()))
+            .ReturnsAsync(GetTeamList());
+
+        _mockMapper
+            .Setup(x => x
+            .Map<IEnumerable<TeamMemberDTO>>(It.IsAny<IEnumerable<TeamMember>>()))
+            .Returns(GetListTeamDTO());
+
+        var handler = new GetAllTeamHandler(_mockRepository.Object, _mockMapper.Object);
+
+        //Act
+        var result = await handler.Handle(new GetAllTeamQuery(), CancellationToken.None);
+
+        //Assert
+        Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.Equal(GetTeamList().Count(), result.Value.Count())
```

Comment:
the same is here because the result can be not null, but the value can, so you will cat null reference exception in the test